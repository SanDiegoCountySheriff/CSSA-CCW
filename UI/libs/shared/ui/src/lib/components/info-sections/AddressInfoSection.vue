<template>
  <v-container
    fluid
    class="address-info-container rounded"
  >
    <v-banner class="font-weight-bold text-xl text-left mb-5">
      {{ $t(props.title) }}
    </v-banner>

    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Address 1') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.addressLine1 }}
          </div>
        </div>
      </v-col>

      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Address 2') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.addressLine2 }}
          </div>
        </div>
      </v-col>
    </v-row>

    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('City') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.city }}
          </div>
        </div>
      </v-col>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('State') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.state }}
          </div>
        </div>
      </v-col>
    </v-row>

    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('County') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.county }}
          </div>
        </div>
      </v-col>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Zip') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.zip }}
          </div>
        </div>
      </v-col>
    </v-row>

    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Country') }}
          </v-chip>
          <div class="info-text">
            {{ props.addressInfo.country }}
          </div>
        </div>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import { AddressInfoType } from '@shared-utils/types/defaultTypes';

interface IAddressInfoSectionProps {
  addressInfo: AddressInfoType;
  title: string;
  color: string;
}
const props = defineProps<IAddressInfoSectionProps>();
</script>

<style lang="scss" scoped>
.address-info-container {
  width: 80%;
  height: 100%;
  margin: 0;
  padding: 0;
}
.info-row {
  display: flex;
  flex-direction: row;
}

.info-text {
  margin-left: 0.5rem;
  text-align: start;
  height: 6.5em;
  width: 50%;
  margin-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-top: 0.2rem;
  background-color: rgba(211, 241, 241, 0.3);
  border-bottom: 1px solid #666;
  border-radius: 5px;
  font-size: 1.2em;
  font-weight: bold;
}
</style>
