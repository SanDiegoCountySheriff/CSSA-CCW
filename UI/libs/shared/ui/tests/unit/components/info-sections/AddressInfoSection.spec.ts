/* eslint-disable @typescript-eslint/ban-ts-comment */
import AddressInfoSection from '@shared-ui/components/info-sections/AddressInfoSection.vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'

const localVue = createLocalVue()

describe('AddressInfoSection', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    //@ts-ignore
    wrapper = mount(AddressInfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        addressInfo: {
          addressLine: 'line',
          city: 'city',
          state: 'state',
          county: 'county',
          zip: 'zip',
          country: 'country',
        },
      },
    })
  })

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
