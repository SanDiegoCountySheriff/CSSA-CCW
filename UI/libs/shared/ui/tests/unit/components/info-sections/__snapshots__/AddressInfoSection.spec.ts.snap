// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInfoSection Should match the snapshot 1`] = `
<div class="container address-info-container rounded container--fluid">
  <transition-stub name="expand-transition" class="font-weight-bold text-xl text-left mb-5">
    <div class="v-banner v-sheet theme--light v-banner--is-mobile">
      <div class="v-banner__wrapper">
        <div class="v-banner__content">
          <div class="v-banner__text">

          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Address 1
        </span></span>
        <div class="info-text">
          line1
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Address 2
        </span></span>
        <div class="info-text">
          line2
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          City
        </span></span>
        <div class="info-text">
          city
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          State
        </span></span>
        <div class="info-text">
          state
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          County
        </span></span>
        <div class="info-text">
          county
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Zip
        </span></span>
        <div class="info-text">
          zip
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Country
        </span></span>
        <div class="info-text">
          country
        </div>
      </div>
    </div>
  </div>
</div>
`;
